internal/pkg/providers/azure/azure.go

```diff
@@ -0,0 +1,73 @@
+package azure
+
+import (
+	"github.com/Azure/azure-sdk-for-go/profiles/2019-03-01/network/mgmt/network"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-05-01/resources"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/sirupsen/logrus"
+	"github.com/stakater/Whitelister/internal/pkg/config"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"github.com/stakater/Whitelister/internal/pkg/utils"
+)
+
+// Azure provider class implementing the Provider interface
+type Azure struct {
+	ClientSet                 clientset.Interface
+	resourcesClient           resources.Client
+	securityGroupClient       network.SecurityGroupsClient
+	securityRulesClient       network.SecurityRulesClient
+	SubscriptionID            string
+	ClientID                  string
```

Comment:
We should not keep fields that will not be used in future in this structure, e.g ClientID, ClientSecret, TenantID and subscriptionID.